homeassistant/components/thermostat/nest.py

```diff
@@ -98,7 +107,9 @@ def is_away_mode_on(self):
         return self.structure.away
 
     def set_temperature(self, temperature):
-        """ Set new target temperature """
+        """ Set new target temperature ensuring it is in proper units """
+        temperature = convert(temperature, self.hass.config.temperature_unit,
```

Comment:
I think that instead of calling `convert` here, we should move this call inside [this `for` loop](https://github.com/balloob/home-assistant/blob/master/homeassistant/components/thermostat/__init__.py#L88-L89):
```python
            for thermostat in target_thermostats:
                thermostat.set_temperature(convert(
                    temperature, self.hass.config.temperature_unit, thermostat.unit_of_measurement))
```

That way, it will be solved once and for all.